src/Maker/AbstractDockerMaker.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Maker;
+
+use Symfony\Bundle\MakerBundle\ConsoleStyle;
+use Symfony\Bundle\MakerBundle\DependencyBuilder;
+use Symfony\Bundle\MakerBundle\FileManager;
+use Symfony\Bundle\MakerBundle\Generator;
+use Symfony\Bundle\MakerBundle\Util\ComposeFileManipulator;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * Convenient abstract class for docker makers.
+ *
+ * @author  Jesse Rushlow <jr@rushlow.dev>
+ *
+ * @internal
+ */
+abstract class AbstractDockerMaker extends AbstractMaker
+{
+    protected $fileManager;
+
+    /**
+     * @var ComposeFileManipulator
+     */
+    protected $composeFileManipulator;
+
+    /**
+     * @var string
+     */
+    protected $composeFilePath = '';
+
+    /**
+     * @var bool True if an existing docker-compose.yaml file was found.
+     */
+    protected $existingComposeFile = false;
+
+    /**
+     * @var string Service identifier to be set in docker-compose.yaml.
+     */
+    protected $serviceName = '';
+
+    /**
+     * @var string Version set in docker-compose.yaml for the service. e.g. latest
+     */
+    protected $serviceVersion = '';
+
+    public function __construct(FileManager $fileManager)
+    {
+        $this->fileManager = $fileManager;
+    }
+
+    public function interact(InputInterface $input, ConsoleStyle $io, Command $command): void
```

Comment:
We could refactor this into a `getComposeFileManipulator(ConsoleStyle $io): ComposeFileManipulator` type of function. You would then call it from your `interact` method. We (in this class) would store it on a `private` property and avoid creating it twice.

Reply:
Hmm i like it. But we also call `$this->composeFileManipulator` in the generate method.. it seem that having to `$manipulator = $this->getComposeFileManipulator($io)` twice in different methods of `MakeDockerDatabase` isn't as dry as providing access to `AbstractDockerMaker::composeFileManipulator`.

Perhaps hold off and see how things look with additional makers. Then refactor this to be a private w/ getter vs protected?. I can see a few different ways to tackle this, just not sure which would be the best for the next round of make dockers..

Reply:
what abstract maker?